src/object_store.go

```diff
@@ -0,0 +1,215 @@
+package swift
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects"
+	"github.com/sirupsen/logrus"
+)
+
+// ObjectStore is swift type that holds client and log
+type ObjectStore struct {
+	client *gophercloud.ServiceClient
+	log    logrus.FieldLogger
+}
+
+// NewObjectStore instantiates a Swift ObjectStore.
+func NewObjectStore(log logrus.FieldLogger) *ObjectStore {
+	return &ObjectStore{log: log}
+}
+
+// Authenticate to Swift
+func Authenticate() (*gophercloud.ProviderClient, error) {
+	authOpts, err := openstack.AuthOptionsFromEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	pc, err := openstack.NewClient(authOpts.IdentityEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsVerify, err := strconv.ParseBool(GetEnv("OS_VERIFY", "true"))
+	if err != nil {
+		return nil, fmt.Errorf("Cannot parse boolean from OS_VERIFY environment variable: %v", err)
+	}
+
+	tlsconfig := &tls.Config{}
+	tlsconfig.InsecureSkipVerify = tlsVerify
+	transport := &http.Transport{TLSClientConfig: tlsconfig}
+	pc.HTTPClient = http.Client{
+		Transport: transport,
+	}
+
+	err = openstack.Authenticate(pc, authOpts)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	if err := openstack.Authenticate(pc, authOpts); err != nil {
		return nil, err
	}
```

Reply:
Is this commonly used in other repositories? Where do you have it from :smiley:?

Reply:
Is it worth to us also on this line https://github.com/Lirt/velero-plugin-swift/pull/1/files#diff-62f7305e5b61072e61e630d1f923ef3bR89?

Reply:
dunno, read about it in some refactoring blog and I kind of like it ... up to you if you want to adapt this style :D 

> Is it worth to us also on this line

no, because `object` would be visible only in the if scope.

Reply:
True. I like it.